File: 리트코드/0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Deque<Integer> stack = new ArrayDeque<>();
        int len = temperatures.length;
        int[] answer = new int[len];
        for (int i = 0; i < len; i++) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
                Integer idx = stack.pop();
                answer[idx] = i - idx;
            }
            stack.push(i);
        }
    
        return answer;
    }
}

File: 리트코드/0739-daily-temperatures/0739-daily-temperatures.kt
import java.util.ArrayDeque


class Solution {
    fun dailyTemperatures(temperatures: IntArray): IntArray {
        val stack: Deque<Int> = ArrayDeque()
        val answer = IntArray(temperatures.size)

        for (i in 0 until temperatures.size) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
                val pop = stack.pop()
                answer[pop] = i - pop
            }
            stack.push(i)
        }

        return answer
    }
}